Extensions/Multiplayer/multiplayertools.ts

```diff
@@ -1519,6 +1626,125 @@ namespace gdjs {
+      if (_isJoiningOrStartingAGame) return;
+
+      const _gameId = gdjs.projectData.properties.projectUuid;
+      if (!_gameId) {
+        handleLobbiesError(
+          runtimeScene,
+          'The game ID is missing, the quick join lobby action cannot continue.'
+        );
+        return;
+      }
+
+      _isJoiningOrStartingAGame = true;
+      if (displayLoader) {
+        gdjs.multiplayerComponents.displayLoader(runtimeScene, true);
+      }
+
+      const quickJoinLobbyRelativeUrl = `/play/game/${_gameId}/public-lobby/action/quick-join`;
+
+      try {
+        const quickJoinLobbyResponse: QuickJoinLobbyResponse = await gdjs.evtTools.network.retryIfFailed(
+          { times: 2 },
+          () =>
+            // TODO: should the supported compression methods be sent to the endpoint?
+            fetchAsPlayer({
+              relativeUrl: quickJoinLobbyRelativeUrl,
+              method: 'POST',
+              dev: isUsingGDevelopDevelopmentEnvironment,
+              additionalQueryStringParameters: {
+                isPreview: runtimeScene.getGame().isPreview().toString(),
+                gameVersion: runtimeScene.getGame().getGameData().properties
+                  .version,
+              },
+            })
+        );
+
+        if (quickJoinLobbyResponse.status === 'full') {
+          gdjs.multiplayerComponents.showNotification({
+            runtimeScene,
+            content: 'All lobbies are full.',
+            type: 'error',
+          });
+          return;
+        }
+        if (quickJoinLobbyResponse.status === 'not-enough-players') {
+          // TODO: Should this notification be displayed? The creator might not want it displayed
+          // and open the lobbies iframe instead.
+          gdjs.multiplayerComponents.showNotification({
+            runtimeScene,
+            content: 'No available lobbies, create one.',
+            type: 'error',
+          });
+          return;
+        }
+
+        if (quickJoinLobbyResponse.status === 'join-game') {
+          if (quickJoinLobbyResponse.lobby.status === 'waiting') {
+            _shouldStartGameRightAfterJoiningLobby = true;
+          } else if (quickJoinLobbyResponse.lobby.status === 'playing') {
+            _shouldJoinGameRightAfterJoiningLobby = true;
+          } else {
+            throw new Error(
+              `Lobby in wrong status: ${quickJoinLobbyResponse.status}`
+            );
+          }
+        } else {
+          if (_connectionId) {
+            // Already connected to a lobby.
+            openLobbiesWindow(runtimeScene);
+            return;
+          } else {
+            _shouldOpenLobbyPageRightAfterJoiningLobby = true;
+          }
+        }
+        handleJoinLobbyEvent(runtimeScene, quickJoinLobbyResponse.lobby.id);
+      } catch (error) {
+        logger.error('An error occurred while joining a lobby:', error);
+        onLobbyQuickJoinFinished(runtimeScene);
+      }
+    };
+
+    export const quickJoinLobby = async (
```

Comment:
is there a protection to avoid running that function if a user clicks multiple times? (and avoid opening the auth window multiple times maybe or doing the doQuickJoinLobby?)

Reply:
Good idea, that's done